lib/secret-tasks.js

```diff
@@ -0,0 +1,253 @@
+'use strict';
+
+/**
+ * @module SecretTasks
+ */
+
+var Q = require('q');
+var _ = require('lodash');
+var logger = require('winston');
+var Credstash = require('nodecredstash');
+var yaddaSecret = require('@asymmetrik/yadda-secret');
+
+/**
+ * Generate Credstash credential store
+ * @param {object} options - user provided options for the credential store
+ * @param {object} options.DeploymentCenter - user provided options for the deployment center
+ * @param {string} options.DeploymentCenter.name - Deployment center name
+ * @param {string} options.DeploymentCenter.region - Deployment center region
+ * @param {object} options.DeploymentCenter.secret - Secret options
+ * @param {string} options.DeploymentCenter.secret.kmsKeyAlias - KMS Master Key Alias to encrypt/decrypt data
+ * @return {module.exports|Credstash}
+ */
+function getCredentialStore(options){
+	var tableDetails = getCredentialTableDetails(options);
+	var store = new Credstash({
+		table: tableDetails.name,
+		awsOpts: { region: tableDetails.region },
+		kmsKey: options.DeploymentCenter.secret.kmsKeyAlias,
+	});
+	store.__name = tableDetails.name;
```

Comment:
Why is this line necessary?

Reply:
Just so I can retrieve the name of the table later. It's a double underscore name so it shouldn't be used, if ever.